Plugins/DD4hep/src/DD4hepBinningHelpers.cpp

```diff
@@ -0,0 +1,131 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/DD4hep/DD4hepBinningHelpers.hpp"
+
+void Acts::DD4hepBinningHelpers::decodeBinning(
+    dd4hep::rec::VariantParameters &variantParams, const xml_comp_t &xmlBinning,
+    const std::string &bname, const std::vector<std::string> &bvals) {
+  // Set the surface binninng parameter to true
+  variantParams.set<int>(std::string(bname + "_dim"), bvals.size());
+  for (const auto &bv : bvals) {
+    // Gather the number of bins, 0 indicates variable binning
+    int nBins = Acts::getAttrValueOr<int>(xmlBinning, std::string("n" + bv), 0);
+    // Gather the bin expansion parameter, expansion of 0 is default
+    int nExpansion =
+        Acts::getAttrValueOr<int>(xmlBinning, std::string(bv + "expansion"), 0);
+    // Auto-range detection
+    bool autoRange = Acts::getAttrValueOr<bool>(
+        xmlBinning, std::string(bv + "autorange"), false);
+    variantParams.set<bool>(bname + "_" + bv + "_autorange", autoRange);
+    variantParams.set<int>(bname + "_" + bv + "_exp", nExpansion);
+    // Equidistant binning detected
+    if (nBins > 0) {
+      // Set the type identificatio
+      variantParams.set<std::string>(bname + "_" + bv + "_type", "equidistant");
+      // Set the number of bins
+      variantParams.set<int>(bname + "_" + bv + "_n", nBins);
+      // Set min/max paraeter
+      if (!autoRange) {
+        variantParams.set<double>(
+            bname + "_" + bv + "_min",
+            xmlBinning.attr<double>(std::string(bv + "min").c_str()));
+        variantParams.set<double>(
+            bname + "_" + bv + "_max",
+            xmlBinning.attr<double>(std::string(bv + "max").c_str()));
+      }
+    } else {
+      // Variable binning detected
+      variantParams.set<std::string>(bname + "_" + bv + "_type", "variable");
+      // Get the number of bins explicitly
+      auto boundaries =
+          xmlBinning.attr<std::string>(std::string(bv + "boundaries").c_str());
+      std::string del = ",";
+      int end = boundaries.find(del);
+      int ib = 0;
+      // Unit conversion
+      double unitScalar = 1.;
+      if (bv != "phi") {
+        unitScalar = Acts::UnitConstants::mm / dd4hep::millimeter;
+      }
+      // Split and convert
+      while (end != -1) {
```

Comment:
```suggestion
      while (end != boundaries.end()) {
```

?

Reply:
Updated.

Reply:
You're right, this is an `std::string`. However, can you use `std::string::npos` anyway? This is in fact `-1`, but it's usually used as the end sentinel value for strings.

Reply:
Ah, thanks!